test/tools/ossfuzz/SolidityEvmoneInterface.h

```diff
@@ -0,0 +1,116 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <test/EVMHost.h>
+
+#include <libsolidity/interface/CompilerStack.h>
+
+#include <libyul/AssemblyStack.h>
+
+#include <libsolutil/Keccak256.h>
+
+#include <evmone/evmone.h>
+
+namespace solidity::test::fuzzer
+{
+class SolidityCompilationFramework
+{
+public:
+	explicit SolidityCompilationFramework(langutil::EVMVersion _evmVersion = {});
+
+	Json::Value getMethodIdentifiers()
+	{
+		return m_compiler.methodIdentifiers(m_compiler.lastContractName());
+	}
+	bytes compileContract(
+		std::string const& _sourceCode,
+		std::string const& _contractName,
+		std::map<std::string, solidity::util::h160> const& _libraryAddresses = {},
+		frontend::OptimiserSettings _optimization = frontend::OptimiserSettings::minimal()
+	);
+protected:
+	frontend::CompilerStack m_compiler;
+	langutil::EVMVersion m_evmVersion;
+};
+
+struct EvmoneUtility
```

Comment:
Right, this needs a closer look because semantic tests is platform agnostic while the fuzzer is Linux only. I could do a follow up PR to do further refactoring.

Reply:
Ok